src/stash-list/bookmark.vue

```diff
@@ -56,18 +57,23 @@ export default defineComponent({
 
         related_tabs(): Tab[] {
             if (! this.bookmark.url) return [];
+            const target_window = this.targetWindow;
             const related = this.model().tabs.tabsWithURL(this.bookmark.url);
-            return Array.from(related);
+            return Array.from(related).filter(t => t.windowId === target_window);
         },
 
-        hasOpenTab(): boolean {
-            return !! this.related_tabs.find(
-                t => ! t.hidden && t.windowId === this.targetWindow);
-        },
-        hasActiveTab(): boolean {
-            // TODO look only at the current window
-            return !! this.related_tabs.find(
-                t => t.active && t.windowId === this.targetWindow);
+        hasTabState(): { open: boolean, active: boolean, loading: boolean } {
```

Comment:
`hasTabState` makes me think it's a boolean property, but it's actually a set of booleans; perhaps just call it `tabState`?

Reply:
Changed.